improver/threshold.py

```diff
@@ -396,3 +384,131 @@ def process(self, input_cube: Cube) -> Cube:
+            cube:
+                Cube containing thresholded data (1s and 0s)
+            threshold:
+                Values at which the data has been thresholded (matches cube's y-axis)
+        """
+        coord = iris.coords.AuxCoord(threshold.astype(FLOAT_DTYPE), units=cube.units)
+        coord.rename(self.threshold_coord_name)
+        coord.var_name = "threshold"
+        cube.add_aux_coord(coord, data_dims=len(cube.shape) - 2)
+
+    def process(self, input_cube: Cube) -> Cube:
+        """Convert each point to a truth value based on provided threshold
+        function. If the plugin has a "threshold_units"
+        member, this is used to convert a copy of the input_cube into
+        the units specified.
+
+        Args:
+            input_cube:
+                Cube to threshold. Must have a latitude coordinate.
+
+        Returns:
+            Cube after a threshold has been applied. The data within this
+            cube will contain values between 0 and 1 to indicate whether
+            a given threshold has been exceeded or not.
+
+                The cube meta-data will contain:
+                * Input_cube name prepended with
+                probability_of_X_above(or below)_threshold (where X is
+                the diagnostic under consideration)
+                * Threshold dimension coordinate with same units as input_cube
+                * Threshold attribute ("greater_than",
+                "greater_than_or_equal_to", "less_than", or
+                less_than_or_equal_to" depending on the operator)
+                * Cube units set to (1).
+
+        Raises:
+            ValueError: if a np.nan value is detected within the input cube.
+        """
+        if np.isnan(input_cube.data).any():
+            raise ValueError("Error: NaN detected in input cube data")
+
+        self.threshold_coord_name = input_cube.name()
+
+        cube = input_cube.copy()
+        if self.threshold_units is not None:
+            cube.convert_units(self.threshold_units)
+
+        cube.coord("latitude").convert_units("degrees")
+        threshold_variant = cube.coord("latitude").points
+        threshold_over_latitude = np.array(self.threshold_function(threshold_variant))
+
+        # Add a scalar axis for the longitude axis so that numpy's array-
+        # broadcasting knows what we want to do
+        truth_value = self.comparison_operator["function"](
+            cube.data, np.expand_dims(threshold_over_latitude, 1),
+        )
+
+        truth_value = truth_value.astype(FLOAT_DTYPE)
+
+        if np.ma.is_masked(cube.data):
+            # update unmasked points only
+            cube.data[~input_cube.data.mask] = truth_value[~input_cube.data.mask]
+        else:
+            cube.data = truth_value
+
+        self._add_latitude_threshold_coord(cube, threshold_over_latitude)
+        cube.coord(var_name="threshold").convert_units(input_cube.units)
+
+        for func in self.each_threshold_func:
+            cube = func(cube)
```

Comment:
I think that these lines can now be removed, as each_threshold_func is no longer defined for the `LatitudeDependentThreshold` plugin. Is that correct?